### PR TITLE
[commands] Add Duration parser

### LINKED ISSUE #298: [commands] Add DurationParser
DurationParser will be simple way to get a `java.time.Duration` to be used for many thing game cycle duration punishment duration animation duration etc...

The parser will be simple to use as follow

`1h 30m 59s`

Will be 1 hour and 30 minutes and 59 seconds

Parser will have unit as map of string to base duration

For example
```
{
  "y": Duration.ofDays(365),
  "d": Duration.ofDays(1),
  "h": Duration.ofHours(1),
  "m": Duration.ofMinutes(1),
  "s": Duration.ofSeconds(1),
}
```
We will multiple `value * unut base` and add up all of it

for tab will suggest unit as option after value

### PR DESCRIPTION
**Pull request by ikfir**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation)
- [ ] 🐞 Bug fix (fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)

### 📚 Description
Added Duration parser

simple way to get duration input by giving the parser units and it convert for us to `java.time.Duration`
as follow `10m 30s` will be `Duration.ofMinutes(10).plus(Duration.ofSeconds(30));`

Closes #298

### 🧪 How Has This Been Tested?
unit test

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

* **New Features**
  * Duration parsing for commands with common units (y, w, d, h, m, s).
  * Improved tab-completion that correctly handles parsers which may consume multiple arguments.
  * New timer command demonstrating duration parsing with optional player targeting.
  * New sample compound parser for recipes showcasing nested duration + numeric args.

* **Tests**
  * Added comprehensive tests for duration parsing, tab completion, invalid inputs, recipe parsing, and edge cases.

### CODE REVIEW

File: cocoa-beans-commands/src/main/java/net/apartium/cocoabeans/commands/parsers/DurationParser.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+@ApiStatus.AvailableSince("0.0.44")
+public class DurationParser extends ArgumentParser<Duration> {
+
+    public static final String DEFAULT_KEYWORD = "duration";
+
+    public static final Map<String, Duration> DEFAULT_UNITS = Map.of(
```
Comment (JustNOTRO) [inline cocoa-beans-commands/src/main/java/net/apartium/cocoabeans/commands/parsers/DurationParser.java]: months?

File: cocoa-beans-commands/src/test/java/net/apartium/cocoabeans/commands/parsers/DurationParserTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    @Test
+    void moreComplexStuff() {
+        DurationParser parser = new DurationParser();
+
+        assertParserResult(parser, null, null, args("10m 5s"), new ArgumentParser.ParseResult<>(Duration.ofSeconds(605), 2));
+        assertParserResult(parser, null, null, args("2h 10m 5s"), new ArgumentParser.ParseResult<>(Duration.ofSeconds(7805), 3));
+        assertParserResult(parser, null, null, args("10m 2h 5s"), new ArgumentParser.ParseResult<>(Duration.ofSeconds(7805), 3));
+        assertParserResult(parser, null, null, args("1d 2h 10m 5s"), new ArgumentParser.ParseResult<>(Duration.ofSeconds(94205), 4));
+        assertParserResult(parser, null, null, args("10m 2h 1d 5s"), new ArgumentParser.ParseResult<>(Duration.ofSeconds(94205), 4));
+        assertParserResult(parser, null, null, args("1w 2d 2h 10m 5s"), new ArgumentParser.ParseResult<>(
```
[the remark this answers is not in the document]
Comment (ikfir) [reply cocoa-beans-commands/src/test/java/net/apartium/cocoabeans/commands/parsers/DurationParserTest.java]: It work like that

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG
 ## 0.0.47
+- [commands] Add DurationParser
+- [commands] Fix `CompoundParser` tab completion multiple arguments
+- [commands] Add ArgumentParser support multiple arguments
```
[the remark this answers is not in the document]
Comment (liorsl) [reply CHANGELOG.md]: [commands] Add multiple arguments tab completion support to ArgumentParser

**Review by liorsl — APPROVED** [review]
